### PR TITLE
Updated configuration documentation for new clock algorithm.

### PR DESCRIPTION
**Pull request by davidv1992**
Note: also contains some minor code changes to ensure naming of parameters is spelled correctly and uniformly.

### CODE REVIEW

File: CONFIGURATION.md [hunk 1]
```diff
@@ -45,16 +51,65 @@ The following command line options are available. When an option is not provided
...
+
+##### Nts peers
+
+A peer in `server-nts` mode will use NTS (Network Times Security) to communicate with its server. The server must support NTS. The configuration requires the address of the Key Exchange server (the address of the actual NTP server that ends up being used may be different). For example:
+
+```
+[[peers]]
+mode = "server-nts"
+ke_addr = "time.cloudflare.com:4460"
+certificate = "/path/to/certificates.pem"
```
[the remark this answers is not in the document]
Comment (davidv1992) [reply CONFIGURATION.md:95]: We do, see the table above.

[the remark this answers is not in the document]
Comment (davidv1992) [reply CONFIGURATION.md:95]: I would expect users to set this in more controlled environments, however currently the field doesn't exactly fit that use case (it still also uses the system certificates in that case), but that is probably a separate discussion.

File: CONFIGURATION.md [hunk 2]
```diff
@@ -87,18 +144,64 @@ There are a number of options available to influence how time differences to the
...
-| startup-panic-threshold | No limit forward, 1800 backward | Largest time difference the client is allowed to correct during startup. By default, this is unrestricted as we may be the initial source of time for systems without a hardware backed clock. Value provided is in seconds, set to "inf" to disable checking of jumps. |
-| accumulated-threshold | Disabled | Total amount of time difference the client is allowed to correct using steps whilst running. By default, this is unrestricted. Value provided is in seconds, set to 0 to disable checking of accumulated steps. |
 
-For panic thresholds, asymetric thresholds can be configured, allowing a different sized step going forwards compared to going backwards. This is done by configuring a struct with two values, `forward` and `backward` for the panic threshold.
+The new high performance clock algorithm has more options. Most of these are quite straightforward to understand and can be used to tune the style of time synchronization to the users liking (although the defaults are probably fine for most):
```
[the remark this answers is not in the document]
Comment (davidv1992) [reply CONFIGURATION.md:173]: This is quite some work and I would indeed prefer to defer it to the white paper.

[the remark this answers is not in the document]
Comment (davidv1992) [reply CONFIGURATION.md:173]: Hmm, good question, I have no idea for a name. Do you have suggestions?

File: CONFIGURATION.md [hunk 3]
```diff
@@ -45,16 +51,65 @@ The following command line options are available. When an option is not provided
...
+```
+[[peers]]
+mode = "server-nts"
+ke_addr = "time.cloudflare.com:4460"
+certificate = "/path/to/certificates.pem"
+```
+
+##### Pool
+
+`Pool` mode is a convenient way to configure many NTP servers, without having to worry about individual server's IP addresses.
```
[the remark this answers is not in the document]
Comment (davidv1992) [reply CONFIGURATION.md]: fixed.

File: CONFIGURATION.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (folkertdev) [reply CONFIGURATION.md:173]: something with "kalman"?

**Pull request merged**